### PR TITLE
[v6r21] Dirac API returns directory LFN metadata

### PR DESCRIPTION
**Pull request by atsareg**
This is a PR to fix issue #3955 

BEGINRELEASENOTES

*Interfaces
CHANGE: Dirac.py - getLFNMetadata returns result for both file and directory LFNs

ENDRELEASENOTES

### CODE REVIEW

File: Interfaces/API/Dirac.py [hunk 1]
```diff
@@ -1249,19 +1250,32 @@ def getLfnMetadata(self, lfns, printOutput=False):
...
       self.log.warn('Failed to retrieve file metadata from the catalogue')
-      self.log.warn(repsResult['Message'])
-      return repsResult
+      self.log.warn(fileResult['Message'])
+      return fileResult
+
+    repsResult = fileResult['Value']
+    if repsResult['Failed']:
+      # Some entries can be directories
+      dirs = repsResult['Failed'].keys()
```
Comment (chaen) [inline Interfaces/API/Dirac.py]: why do you need to do the copy here ?

File: Interfaces/API/Dirac.py [hunk 2]
```diff
@@ -1249,19 +1250,32 @@ def getLfnMetadata(self, lfns, printOutput=False):
...
+      dirResult = fc.getDirectoryMetadata(dirs)
+      if not dirResult['OK']:
+        self.log.warn('Failed to retrieve directory metadata from the catalogue')
```
Comment (chaen) [inline Interfaces/API/Dirac.py]: maybe put the message as variable part of the logging  ?

File: Interfaces/API/Dirac.py [hunk 3]
```diff
@@ -1249,19 +1250,32 @@ def getLfnMetadata(self, lfns, printOutput=False):
...
+        self.log.warn(dirResult['Message'])
+        return dirResult
+      for dir in dirResult['Value']['Successful']:
```
Comment (chaen) [inline Interfaces/API/Dirac.py]: `dir` is a reserved keyword

File: Interfaces/API/Dirac.py (hunk 1 above)
Comment (atsareg) [reply Interfaces/API/Dirac.py]: Just to be more readable

**Pull request merged**